src/client-agent/start_agent.c

```diff
@@ -51,66 +51,57 @@ bool connect_server(int server_id, bool verbose)
 
         if (agt->server[agt->rip_id].rip) {
             if (verbose) {
-                minfo("Closing connection to server (%s:%d/%s).",
+                minfo("Closing connection to server ([%s]:%d/%s).",
                     agt->server[agt->rip_id].rip,
                     agt->server[agt->rip_id].port,
                     agt->server[agt->rip_id].protocol == IPPROTO_UDP ? "udp" : "tcp");
             }
         }
     }
 
-    char *tmp_str;
-
-    /* Check if we have a hostname */
-    tmp_str = strchr(agt->server[server_id].rip, '/');
+    char *ip_address = NULL;
+    char *tmp_str = strchr(agt->server[server_id].rip, '/');
     if (tmp_str) {
-        /* Resolve hostname */
-        if (!isChroot()) {
-            resolve_hostname(&agt->server[server_id].rip, 5);
-
-            tmp_str = strchr(agt->server[server_id].rip, '/');
-            if (tmp_str) {
-                tmp_str++;
-            }
-        } else {
-            tmp_str++;
-        }
-    } else {
-        tmp_str = agt->server[server_id].rip;
+        // server address comes in {hostname}/{ip} format
+        ip_address = strdup(++tmp_str);
+    }
+    if (!ip_address) {
```

Comment:
Could it be an `else`?
```
if (tmp_str) {
       ip_address = strdup(++tmp_str);
} else {
        // server address is either a host or a ip
        ip_address = OS_GetHost(agt->server[server_id].rip, 3);
}
```

Server addres could be `{hostname}/` ?

Reply:
Yes, it could be anything since it comes from the user configuration input.